ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java

```diff
@@ -64,34 +65,64 @@ public static AbstractStreamCreateStatement checkAndSetAvroSchema(
+      streamsProperties.put(DdlConfig.AVRO_SCHEMA, avroSchemaString);
+      AbstractStreamCreateStatement abstractStreamCreateStatementCopy = addAvroFields(
+              abstractStreamCreateStatement,
+              AvroSchemaTranslator.toKsqlSchema(avroSchemaString),
+              schemaMetadata.getId()
+      );
+      return abstractStreamCreateStatementCopy;
 
-        String avroSchemaString = schemaMetadata.getSchema();
-        streamsProperties.put(DdlConfig.AVRO_SCHEMA, avroSchemaString);
-        AbstractStreamCreateStatement abstractStreamCreateStatementCopy = addAvroFields(
-            abstractStreamCreateStatement,
-            AvroSchemaTranslator.toKsqlSchema(avroSchemaString),
-            schemaMetadata.getId()
-        );
-        return abstractStreamCreateStatementCopy;
-      } else {
-        return abstractStreamCreateStatement;
-      }
-    } catch (Exception e) {
-      String errorMessage = String.format(
-          " Unable to verify the AVRO schema is compatible with KSQL. %s ",
-          e.getMessage()
+    } catch (final Exception e) {
+      throw new KsqlException("KSQL does not currently support the Avro schema for topic "+kafkaTopicName+
+              ".\nReason: " + e.getMessage()+"\n\n"+
+              "Please see https://github.com/confluentinc/ksql/issues/ to see if this particular reason is already\n"+
+              "known, and if not log a new issue. Please include the full Avro schema that you are trying to use."
       );
-      throw new KsqlException(errorMessage);
+    }
+  }
+
+  private static SchemaMetadata getSchemaMetadata(final AbstractStreamCreateStatement abstractStreamCreateStatement,
+                                                  final SchemaRegistryClient schemaRegistryClient,
+                                                  final String kafkaTopicName) {
+    try {
+      return fetchSchemaMetadata(
+              abstractStreamCreateStatement,
+              schemaRegistryClient,
+              kafkaTopicName
+      );
+    } catch (final RestClientException e) {
+      if (e.getStatus() == HttpStatus.SC_NOT_FOUND) {
+        throw new KsqlException("Avro schema for message values on topic " + kafkaTopicName +
+                " does not exist in the Schema Registry.\n"+
+                "Subject: " + kafkaTopicName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX + "\n\n" +
+                "Possible causes include:\n" +
+                "- The topic itself does not exist\n"+
+                "\t-> Use SHOW TOPICS; to check\n" +
+                "- Messages on the topic are not Avro serialized\n" +
+                "\t-> Use PRINT '"+ kafkaTopicName+"' FROM BEGINNING; to verify\n" +
+                "- Messages on the topic have not been serialized using the Confluent Schema Registry Avro serializer\n" +
+                "\t-> See https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html\n" +
```

Comment:
Can we make the links in the error messages constants accessible from the whole codebase? That way if we need to change the links for whatever reason we can do so easily.

Reply:
That's beyond the scope of what I can do in this PR. Is it a blocker?

Reply:
It should just be a matter of adding a static member to https://github.com/confluentinc/ksql/blob/master/ksql-common/src/main/java/io/confluent/ksql/util/KsqlConstants.java and then referencing it here.

Reply:
Done :)